test/extensions/filters/http/ext_proc/streaming_integration_test.cc

```diff
@@ -47,6 +47,8 @@ class StreamingIntegrationTest : public HttpIntegrationTest,
     // This enables a built-in automatic upstream server.
     autonomous_upstream_ = true;
     proto_config_.set_allow_mode_override(true);
+    proto_config_.mutable_message_timeout()->set_seconds(2);
```

Comment:
This still seems likely to flake when overloaded. Is there a way to run these tests with mock/fake time?

Reply:
Yeah, that's doable.  I am thinking to convert all of these tests in streaming_integration_test.cc into unit tests and put them in filter_test.cc.  These will be done in follow up PR.

Reply:
Also *adding* a unit test doesn't really address Harvey's comment that this test would be likely to flake :)

Reply:
Setting failure_mode_allow to true as below will be able to get the tests passing even timeout or gRPC channel random close happens:
+ proto_config_.set_failure_mode_allow(true);

The request will be forwarded as if the ext_proc filter does not exist in this case.

Reply:
IDK what impact that setting has on the realism of the test. At first glance it makes me wonder if the test just isn't really checking that much.

The purpose of simulated time tests is to enable robust predictable testing of code with timeouts, including in integration tests. You can force a timeout to occur,, or not occur, based on explicit control of time from the test fixture.

Reply:
Thanks for the comments! There are other ext_proc tests to test timeout behavior, like here and other tests around it:
https://github.com/envoyproxy/envoy/blob/5c9cc8460c582f1a895dbffd974e248f0b2911bc/test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc#L1690C56-L1690C56

This streaming_integration_test.cc is not to test timeout scenario. It's to test the ext_proc server mutation behaviors when the ext_proc filter is configured with STREAMED or BUFFERED_PARTIAL mode. 

The original test cases are written with the assumption that timeout won't happen and stubby RPC channel will not close. But in the tests, both of them may happen, which leads to flakiness of the test.  So, extending the timeout value from 200ms to 2s will reduce the timeout probability.  Set failure_mode_allow to true will catch up if either errors happens, the tests can still pass, like client still receive 200 instead of 500, et.al.

Reply:
Depending on how flaky this file still is after the proposed change, we can consider convering these streamed integration tests to unit tests using mocks, similar to what we have don in this PR: https://github.com/envoyproxy/envoy/pull/29940.